src/netlink.c

```diff
@@ -230,8 +230,29 @@ netlink_link_add_bridge(struct netlink_handle *const hndl,
 	mnl_attr_put_str(nlh, IFLA_IFNAME, name);
 	link_attr = mnl_attr_nest_start(nlh, IFLA_LINKINFO);
 	mnl_attr_put_str(nlh, IFLA_INFO_KIND, "bridge");
-	mnl_attr_nest_end(nlh, link_attr);
 
+	/* Disable multicast snooping on the bridge.
+	 * Netlink support for this was added in kernel 4.4 where
+	 * IFLA_BR_MCAST_SNOOPING was defined with value 23 in an enum.
+	 * Reference:
+	 * https://github.com/torvalds/linux/blob/master/include/uapi/linux/if_link.h
+	 *
+	 * So check if the last value of the enum is greater than 23, before
+	 * adding the value to the netlink nested attributes to be compatible
+	 * with earlier kernels.
+	 */
+	#ifdef IFLA_BR_MAX
```

Comment:
@dlespiau Checked this with a kernel with CONFIG_BRIDGE_IGMP_SNOOPING disabled. In that case the attribute related to multicast snooping is ignored.
If the bridge or br_netfilter module is disabled, then we would not be able to create the bridge in the first place.
Also note, I am checking for the value of IFLA_BR_MAX at runtime, since this macro is not evaluated at preprocessor time as it depends on an enum value.